From 88f50846d5236baea909f2100847d9091b16233e Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Thu, 25 Jan 2024 11:47:58 +0100
Subject: [PATCH 0623/1695] plat-stm32mp2: update rstctrl driver flag for
 STM32MP21

Add CFG_STM32MP21_RSTCTRL in conf.mk file that will be set when
CFG_STM32MP21 is set. Else, set CFG_STM32MP25_RSTCTRL flag.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 77c01277cc6aa7a57b426dcb68795ad92cbb3e85 plat-stm32mp2: update rstctrl driver flag for STM32MP21]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I9d11601094406b1d502e217a7aedfeab91fe3961
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/361573
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index bf58536b5..b5fb5df81 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -86,13 +86,14 @@ $(call force,CFG_STM32_PWR,y)
 $(call force,CFG_STM32_PWR_REGUL,y)
 $(call force,CFG_STM32MP_CLK_CORE,y)
 $(call force,CFG_STM32MP_REMOTEPROC,y)
-$(call force,CFG_STM32MP25_RSTCTRL,y)
 $(call force,CFG_WITH_ARM_TRUSTED_FW,y)
 $(call force,CFG_WITH_LPAE,y)
 
 ifeq ($(CFG_STM32MP21),y)
+$(call force,CFG_STM32MP21_RSTCTRL,y)
 $(call force,CFG_STM32MP21_CLK,y)
 else
+$(call force,CFG_STM32MP25_RSTCTRL,y)
 $(call force,CFG_STM32MP25_CLK,y)
 endif
 
@@ -188,6 +189,10 @@ CFG_WDT_SM_HANDLER ?= $(CFG_WDT)
 CFG_WDT_SM_HANDLER_ID ?= 0xbc000000
 
 # Enable reset control
+ifeq ($(CFG_STM32MP21_RSTCTRL),y)
+$(call force,CFG_DRIVERS_RSTCTRL,y)
+$(call force,CFG_STM32_RSTCTRL,y)
+endif
 ifeq ($(CFG_STM32MP25_RSTCTRL),y)
 $(call force,CFG_DRIVERS_RSTCTRL,y)
 $(call force,CFG_STM32_RSTCTRL,y)
-- 
2.47.2

